main.go

```diff
@@ -0,0 +1,280 @@
+			log.Warnf("Failed to close response body, error: %s", err)
+		}
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("unsuccessful status code: %d", resp.StatusCode)
+	}
+
+	tmpPath, err := pathutil.NormalizedOSTempDirPath("flank-bin")
+	if err != nil {
+		return "", err
+	}
+
+	binPath := filepath.Join(tmpPath, "flank.jar")
+
+	bodyData, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	return binPath, fileutil.WriteBytesToFile(binPath, bodyData)
+}
+
+func failf(format string, args ...interface{}) {
+	log.Errorf(format, args...)
+	os.Exit(1)
+}
+
+// lists all dirs inside of ./results dir and selects the latest(by modtime)
+// then all the files in the root will be copied from this dir to the root of the dir under BITRISE_DEPLOY_DIR
+func exportArtifacts(copied func(src, dest string)) error {
+	fInfs, err := ioutil.ReadDir(resultsDir)
+	if err != nil {
+		return err
+	}
+
+	var latestDir string
+	var latestModtime time.Time
+
+	for _, fInf := range fInfs {
+		if !fInf.IsDir() {
+			continue
+		}
+		if fInf.ModTime().After(latestModtime) || fInf.ModTime().Equal(latestModtime) {
+			latestModtime = fInf.ModTime()
+			latestDir = filepath.Join(resultsDir, fInf.Name())
+		}
+	}
+
+	fInfs, err = ioutil.ReadDir(latestDir)
+	if err != nil {
+		return err
+	}
+
+	for _, fInf := range fInfs {
+		if fInf.IsDir() {
+			continue
+		}
+
+		srcFile := filepath.Join(latestDir, fInf.Name())
+		destinationFile := filepath.Join(os.Getenv("BITRISE_DEPLOY_DIR"), fInf.Name())
+
+		data, err := ioutil.ReadFile(srcFile)
+		if err != nil {
+			return err
+		}
+		err = ioutil.WriteFile(destinationFile, data, 0644)
+		if err != nil {
+			return err
+		}
+
+		if copied != nil {
+			copied(srcFile, destinationFile)
+		}
+	}
+
+	return nil
+}
+
+func main() {
+	//
+	// configuration
+	var cfg config
+	if err := stepconf.Parse(&cfg); err != nil {
+		failf("Issue with input: %s", err)
+	}
+	stepconf.Print(cfg)
+	fmt.Println()
+
+	//
+	// tool setup
+	log.Infof("Downloading binary")
+	downloadURL, err := getDownloadURLbyVersion(cfg.Version)
+	if err != nil {
+		failf("Failed to get download URL, error: %s", err)
+	}
+
+	binaryPath, err := download(downloadURL)
+	if err != nil {
+		failf("Failed to download binary, error: %s", err)
+	}
+
+	log.Donef("- Done")
+	fmt.Println()
+
+	//
+	// running the tool
+	log.Infof("Running test")
+	platform, err := detectPlatform(cfg.ConfigPath)
+	if err != nil {
+		failf("Failed to detect platform, error: %s", err)
+	}
+	log.Printf("- Detected platform: %s", platform)
+
+	commandFlags, err := shellquote.Split(cfg.CommandFlags)
+	if err != nil {
+		failf("Failed to split command flags, error: %s", err)
+	}
+
+	fmt.Println()
+	command := command.New("java", append([]string{"-jar", binaryPath, platform, "run", "-c", cfg.ConfigPath}, commandFlags...)...).
+		SetStdin(os.Stdin).
+		SetStdout(os.Stdout).
+		SetStderr(os.Stderr)
+
+	log.Donef("$ %s", command.PrintableCommandArgs())
```

Comment:
Logging done before actually running the command is confusing when debugging.

Reply:
We usually print commands that will be run with green and starting with a `$`.